addon/components/frost-select.js

```diff
@@ -240,15 +240,18 @@ export default Component.extend(PropTypeMixin, {
     .fromPairs()
     .value()
 
-    this.get('reduxStore').dispatch(resetDropDown(reduxAttrs))
-
     const dataChanged = isAttrDifferent(newAttrs, oldAttrs, 'data')
     const selectedChanged = isAttrDifferent(newAttrs, oldAttrs, 'selected')
     const selectedValueChanged = isAttrDifferent(newAttrs, oldAttrs, 'selectedValue')
 
+    if (!(selectedChanged || selectedValueChanged)) {
```

Comment:
I find that `if (!selectedChanged && !selectedValueChanged) {` reads a bit better as it doesn't have the extra parenthesis that the negation applies to.